### ISSUE TITLE
different result over running different version of tensorflow

### ISSUE THREAD

**Issue by benam2** [hugochan/KATE#24]
Hi again,
Finally I got the reason why I can not replicate your result. until now I was using `tensorfloe 1.14`. which in this version we get 69 percent accuracy on `20 newsgorup data set` with 512 nd.

However, once I changed the tf version back to 1.2.1 it replicate the same result as reported in paper.

I appreciate it if you can help with this part, why that much difference with only changing the version of tensorflow!

Thanks.

**Comment by hugochan** [comment]
> Hi again,
> Finally I got the reason why I can not replicate your result. until now I was using `tensorfloe 1.14`. which in this version we get 69 percent accuracy on `20 newsgorup data set` with 512 nd.
> 
> However, once I changed the tf version back to 1.2.1 it replicate the same result as reported in paper.
> an example of extracted topics with version 1.2.1
> 
> ```
> line subject organ armenian isra turkish israel armenia arab turkey
> 
> line subject organ car bike speed brake truck ride motorcycl
> 
> line organ subject clipper escrow chip livesey isc cramer key
> 
> line subject organ cramer optilink gay clayton homosexu gun crimin
> 
> line subject organ april space san sexual washington cal ron
> ```
> 
> I appreciate it if you can help with this part, why that much difference with only changing the version of tensorflow!
> 
> Thanks.

Thank you for reporting this. One quick check I would suggest is to load and test the model pretraind using the original TF version in the new TF version to see if you can reproduce the results. 

If not, it means that even in the inference time, the two versions produce different results.  In this case, you might want to further find which function call makes the difference. 

If yes, then maybe the training dynamics or weight initialization in the new version is different from the old version. And you might want to dig it deeper. 

Hope this helps.
